File: src/main/java/com/urbanhunt/app/model/PrizeConfirmation.java
package com.urbanhunt.app.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PrizeConfirmation {

    private String id;
    private String challengeId;
    private String userId; // User who found the prize
    private ConfirmationStatus status;
    private String message; // Optional text message
    private String contentUrl; // Optional photo or video URL
    private Date createdAt;
    private Date confirmedAt; // When status changed to DONE

    public enum ConfirmationStatus {
        NEW,
        DONE
    }
}

File: app/ios/UrbanHunt/Models/PrizeConfirmation.swift
//
//  PrizeConfirmation.swift
//  UrbanHunt
//
//  Model for prize confirmation
//

import Foundation

struct PrizeConfirmation: Codable, Identifiable {
    let id: String
    let challengeId: String
    var userId: String?
    var status: ConfirmationStatus
    var message: String?
    var contentUrl: String?
    let createdAt: Date
    var confirmedAt: Date?

    enum ConfirmationStatus: String, Codable {
        case new = "NEW"
        case done = "DONE"
    }
}